electron/main/ipcEvents.ts

```diff
@@ -69,28 +70,60 @@ export default class IpcEvents {
 
     private vanish() {
         ipcMain.on(IpcEvent.Vanish, () => {
-            log.info('Vanishing overlay');
-
-            const windowBounds = this.overlay?.getBounds() as Rectangle;
-
-            log.info('Saved window state', windowBounds);
-
-            const theme = this.store.get('savedWindowState').theme;
-
-            const data: WindowState = {
-                x: windowBounds.x,
-                y: windowBounds.y,
-                width: windowBounds.width,
-                height: windowBounds.height,
-                isClickThrough: true,
-                isTransparent: true,
-                theme,
-            };
-
-            this.store.set('savedWindowState', data);
+            if (!this.store.get('settings').isOpen && this.store.get('savedWindowState').isTransparent && this.store.get('chatOptions').channel != '') {
+                // DISABLING VANISH CASE
+                // Settings are CLOSED and the Window IS transparent
+                log.info('Disabling Vanish');
+                this.store.set<typeof StoreKeys.SavedWindowState>('savedWindowState', {
+                    ...this.store.get('savedWindowState'),
+                    isClickThrough: false,
+                    isTransparent: false,
+                });
+
+                this.overlay?.setIgnoreMouseEvents(false);
+                this.overlay?.webContents.send(IpcEvent.ShowApp);
+            } else if (!this.store.get('settings').isOpen && !this.store.get('savedWindowState').isTransparent && this.store.get('chatOptions').channel != '') {
```

Comment:
as mentioned here https://github.com/Enubia/ghost-chat/pull/773#discussion_r1457366176 we should probably not do that right now since now the user cannot vanish the window anymore if they use some external sources

I will think about a way to do that and we will keep it as it was before so that they can vanish the app anywhere for now

Reply:
reverted it